File: app/src/main/java/com/baidu/tv/player/model/TokenResponse.java
package com.baidu.tv.player.model;

import com.google.gson.annotations.SerializedName;

/**
 * Token响应模型
 */
public class TokenResponse {
    
    @SerializedName("access_token")
    private String accessToken;
    
    @SerializedName("refresh_token")
    private String refreshToken;
    
    @SerializedName("expires_in")
    private long expiresIn;
    
    @SerializedName("scope")
    private String scope;
    
    @SerializedName("session_key")
    private String sessionKey;
    
    @SerializedName("session_secret")
    private String sessionSecret;
    
    @SerializedName("error")
    private String error;
    
    @SerializedName("error_description")
    private String errorDescription;

    // Getters and Setters
    public String getAccessToken() {
        return accessToken;
    }

    public void setAccessToken(String accessToken) {
        this.accessToken = accessToken;
    }

    public String getRefreshToken() {
        return refreshToken;
    }

    public void setRefreshToken(String refreshToken) {
        this.refreshToken = refreshToken;
    }

    public long getExpiresIn() {
        return expiresIn;
    }

    public void setExpiresIn(long expiresIn) {
        this.expiresIn = expiresIn;
    }

    public String getScope() {
        return scope;
    }

    public void setScope(String scope) {
        this.scope = scope;
    }

    public String getSessionKey() {
        return sessionKey;
    }

    public void setSessionKey(String sessionKey) {
        this.sessionKey = sessionKey;
    }

    public String getSessionSecret() {
        return sessionSecret;
    }

    public void setSessionSecret(String sessionSecret) {
        this.sessionSecret = sessionSecret;
    }

    public String getError() {
        return error;
    }

    public void setError(String error) {
        this.error = error;
    }

    public String getErrorDescription() {
        return errorDescription;
    }

    public void setErrorDescription(String errorDescription) {
        this.errorDescription = errorDescription;
    }

    /**
     * 是否有错误
     */
    public boolean hasError() {
        return error != null && !error.isEmpty();
    }
}

File: app/src/main/java/com/baidu/tv/player/kt/model/TokenResponse.kt
package com.baidu.tv.player.kt.model

import com.google.gson.annotations.SerializedName

/**
 * Token 响应模型（设备码换 token / 刷新 token 共用）。
 */
data class TokenResponse(
    @SerializedName("access_token") val accessToken: String? = null,
    @SerializedName("refresh_token") val refreshToken: String? = null,
    @SerializedName("expires_in") val expiresIn: Long = 0,
    @SerializedName("scope") val scope: String? = null,
    @SerializedName("session_key") val sessionKey: String? = null,
    @SerializedName("session_secret") val sessionSecret: String? = null,
    @SerializedName("error") val error: String? = null,
    @SerializedName("error_description") val errorDescription: String? = null,
) {
    /** 是否有错误。 */
    fun hasError(): Boolean = !error.isNullOrEmpty()
}
